cmd/kubeadm/app/cmd/config.go

```diff
@@ -571,11 +585,45 @@ type ImagesList struct {
 	cfg *kubeadmapi.InitConfiguration
 }
 
+// imageTextPrinter prints image info in a text form
+type imageTextPrinter struct {
+	output.TextPrinter
+}
+
+// PrintObj is an implementation of ResourcePrinter.PrintObj for plain text output
+func (itp *imageTextPrinter) PrintObj(obj runtime.Object, writer io.Writer) error {
+	for _, img := range obj.(*outputapiv1alpha1.Images).Images {
+		if _, err := fmt.Fprintf(writer, "%s/%s:%s\n", img.Prefix, img.Name, img.Tag); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// imageTextPrintFlags provides flags necessary for printing image in a text form.
+type imageTextPrintFlags struct{}
+
+// ToPrinter returns kubeadm printer for the text output format
+func (ipf *imageTextPrintFlags) ToPrinter(outputFormat string) (output.Printer, error) {
+	if outputFormat == output.TextOutput {
+		return &imageTextPrinter{}, nil
+	}
+	return nil, genericclioptions.NoCompatiblePrinterError{OutputFormat: &outputFormat, AllowedFormats: []string{output.TextOutput}}
+}
+
 // Run runs the images command and writes the result to the io.Writer passed in
-func (i *ImagesList) Run(out io.Writer) error {
+func (i *ImagesList) Run(out io.Writer, printer output.Printer) error {
 	imgs := images.GetControlPlaneImages(&i.cfg.ClusterConfiguration)
+
+	var outImages []outputapiv1alpha1.Image
 	for _, img := range imgs {
-		fmt.Fprintln(out, img)
+		splitted := strings.SplitN(img, "/", 2)
```

Comment:
should be just `split`. `split` is the past tense of `split`.

Reply:
this requires error handling to check if the slice len is < 2.
[1]

Reply:
It can't be < 2 in current implementation. image is formatted by [this code](https://github.com/kubernetes/kubernetes/blob/release-1.17/cmd/kubeadm/app/images/images.go#L31).

I agree with @rosti here. It should be done differently to avoid parsing the string, which was just formatted. However, this seem to be out of scope of this PR.

Reply:
Done. Please review again.